src/cargo/ops/cargo_compile.rs

```diff
@@ -235,7 +235,17 @@ pub fn compile_ws<'a>(
         ref target_rustc_args,
     } = *options;
 
-    let target = target.clone();
+    let target = match target {
+        &Some(ref target) if target.ends_with(".json") => {
```

Comment:
Is this clause required here? Ideally there'd only be one location where we'd deal with json targets and currently it's split across two locations

Reply:
I'm not sure which other location you mean. The `Layout::new` implementation?

The motivation for putting it here is that we canonicalize the path that is passed to `rustc`, to avoid the issue that the path becomes invalid due to the cwd changes.

Reply:
Oh sure I didn't look too too closely before, I was mostly just thinking that we should have one location where we recognize `.json` targets instead of two, but the most recent version only has one so that sounds good to me!